src/main/docs/guide/httpServer/transfers.adoc

```diff
@@ -38,6 +38,31 @@ The server supports returning `304` (Not Modified) responses if the files being
 
 TIP: To use a custom data source to send data through an input stream, construct a link:{jdkapi}/java.base/java/io/PipedInputStream.html[PipedInputStream] and link:{jdkapi}/java.base/java/io/PipedOutputStream.html[PipedOutputStream] to write data from the output stream to the input. Make sure to do the work on a separate thread so the file can be returned immediately.
 
+== Sending Reactive Streams as File Downloads
+Micronaut also supports returning *reactive streams* (e.g., `Flux`, `Flowable`,
+or any `Publisher`) without buffering the entire response in memory. If you want to
+force the client browser to download the streamed data (for example, CSV lines),
+you can set the `Content-Disposition: attachment` header.
+
+[source,java]
+----
+@Get("/csv")
+public HttpResponse<Flux<String>> downloadCsv() {
+    Flux<String> data = Flux.just(
+        "data1,data2",
+        "data3,data4"
+    );
+    return HttpResponse.ok(data)
+        .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"data.csv\"")
+        .contentType(MediaType.TEXT_PLAIN_TYPE);
+}
```

Comment:
could you move this to a source file that compiles that preferably has a test and use the `snippet::` macro to include with examples for Java, Kotlin and Groovy. Thanks.